cdk/v2/handler.js

```diff
@@ -1,27 +1,51 @@
-const { WorkflowUtils, WorkflowEngine } = require("rudder-workflow-engine");
+const { WorkflowEngineFactory } = require("rudder-workflow-engine");
+const logger = require("../../logger");
+
+const ErrorBuilder = require("../../v0/util/error");
 const {
   getRootPathForDestination,
   getWorkflowPath,
   getPlatformBindingsPaths
 } = require("./utils");
 
-const destinationWorkflowEngineMap = new Map();
-
-async function getWorkflowEngine(destName, flowType) {
-  // Create a new instance of the engine for the destination if needed
-  // TODO: Use cache to avoid long living engine objects
-  if (!destinationWorkflowEngineMap.has(destName)) {
+async function getWorkflowEngineInternal(destName, flowType) {
+  try {
     const destRootDir = getRootPathForDestination(destName);
     const workflowPath = await getWorkflowPath(destRootDir, flowType);
-
-    const workflowEngine = new WorkflowEngine(
-      WorkflowUtils.createFromFilePath(workflowPath),
+    const platformBindingsPaths = await getPlatformBindingsPaths();
+    return WorkflowEngineFactory.createFromFilePath(
+      workflowPath,
       destRootDir,
-      await getPlatformBindingsPaths()
+      platformBindingsPaths
+    );
+  } catch (error) {
```

Comment:
Should we handle the errors from engine lib separately? By throwing a new error, we are losing the original one.

Reply:
Sure we can remove from here.